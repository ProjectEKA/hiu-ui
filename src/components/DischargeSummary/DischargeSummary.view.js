import React from 'react';
import * as PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import useStyle from './DischargeSummary.style';
import { formatDateString } from '../common/HealthInfo/FhirResourcesUtils';

const DischargeSummary = ({ title, startDate, endDate, authors, status, date }) => {
  const classes = useStyle();
  const start = formatDateString(startDate);
  const end = formatDateString(endDate);
  return (
    title ? (
      <Grid container className={classes.container} justify='space-between'>
        <Grid item>
          <Typography variant='h6' color='inherit' className={classes.title}>
            Document: 
            {' '}
            {title}
          </Typography>
        </Grid>
        <Grid item className={classes.description}>
          {date && (
          <div>
            <span className={classes.label}>Date: </span>
            {date}
          </div>
)}
          {start && end && (
          <span className={classes.period}>
            Admitted from
            {start}
            {' '}
            to
            {end}
          </span>
)}
          {authors.length && (
          <div>
            <span className={classes.label}>Authors: </span>
            { authors.join(', ') }
          </div>
)}
          {status && (
          <div>
            <span className={classes.label}>Status: </span>
            {status.toUpperCase()}
          </div>
)}
        </Grid>
      </Grid>
    ) : ''
  );
};

DischargeSummary.propTypes = {
  title: PropTypes.string,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  authors: PropTypes.arrayOf(PropTypes.string),
  status: PropTypes.string,
  date: PropTypes.string,
};

DischargeSummary.defaultProps = {
  title: '',
  startDate: '',
  endDate: '',
  authors: [],
  status: '',
  date: '',
};


export default DischargeSummary;
