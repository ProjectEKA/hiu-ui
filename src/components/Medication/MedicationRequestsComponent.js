import React from 'react';
import * as PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { formatDateString } from '../common/HealthInfo/FhirResourcesUtils';
import TableStyles from '../common/Styles/Table.style';
import MedicationPriority from './MedicationPriority';
import MedicationNote from './MedicationNote';
import MedicationDose from './MedicationDose';


const displayCodeableConcept = (codeableConcept, defaultText) => {
  if (codeableConcept.text) {
    return codeableConcept.text;
  }
  if (codeableConcept.coding) {
    return codeableConcept.coding[0].display
      ? codeableConcept.coding[0].display
      : codeableConcept.coding[0].code;
  }
  return defaultText;
};

const findMedicationName = (mr) => {
  const medicationConcept 
    = (mr.medicationReference && mr.medicationReference.targetResource) 
      ? mr.medicationReference.targetResource.code 
      : mr.medicationCodeableConcept;
  if (!medicationConcept) return "Unspecified";
  return displayCodeableConcept(medicationConcept, 'Unspecified');
};

  
const MedicationReason = ({reasons, reasonCodes}) => {
  if (reasons) {
    return (
      <li>
        Reasons:&nbsp;
        <span>
          {reasons.map((ref) => displayCodeableConcept(ref.targetResource.code, '')).reduce((acc, value) => `${acc}, ${value}`)}
        </span>
      </li>
    );
  }

  if (reasonCodes) {
    return (
      <li>
        Reasons:&nbsp;
        <span>
          {reasonCodes.map((code) => displayCodeableConcept(code, '')).reduce((acc, value) => `${acc}, ${value}`)}
        </span>
      </li>
    );
  }
  return null;
};

// eslint-disable-next-line max-len
const MedicationRequestsComponent = ({ medicationRequests }) => (medicationRequests && medicationRequests.length > 0 ? (
  <TableStyles>
    <TableContainer className="table-container" component={Paper}>
      <Table className="table" aria-label="simple table">
        <TableHead>
          <TableRow className="table-head">
            <TableCell className="header" colSpan={4}>
              Medication:
            </TableCell>
          </TableRow>
          <TableRow className="table-head">
            <TableCell align="left">Date</TableCell>
            <TableCell align="left">Medication</TableCell>
            <TableCell align="left">Dosing Instruction</TableCell>
            <TableCell align="left">Additional Info</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {medicationRequests.map((mr) => (
            <TableRow key={mr.authoredOn}>
              <TableCell className="table-cell">
                {mr.authoredOn ? formatDateString(mr.authoredOn) : ''}
              </TableCell>
              <TableCell className="table-cell">
                {findMedicationName(mr)}
                {` (${mr.status})`}
                <ul className="mediation-list-item">
                  {<MedicationReason reasons={mr.reasonReference} reasonCodes={mr.reasonCode}/>}
                </ul>
              </TableCell>
              <TableCell className="table-cell">
                <MedicationDose dosageInstructions={mr.dosageInstruction} />
              </TableCell>
              <TableCell className="table-cell">
                <ul className="mediation-list-item">
                  {<MedicationPriority mr={mr} />}
                  {<MedicationNote mr={mr} />}
                </ul>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  </TableStyles>
) : (
  <div />
));

const medicationRequestShape = {
  status: PropTypes.string,
  authoredOn: PropTypes.string,
  dosageInstruction: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
  })),
  medicationReference: PropTypes.shape({
    targetResource: PropTypes.shape({
      code: PropTypes.shape({
        coding: PropTypes.arrayOf(PropTypes.shape({
          code: PropTypes.string,
          display: PropTypes.string,
        })),
      }),
    }),
  }),
};

MedicationRequestsComponent.propTypes = {
  medicationRequests: PropTypes.arrayOf(PropTypes.shape(medicationRequestShape)),
};

MedicationRequestsComponent.defaultProps = {
  medicationRequests: [],
};

export default MedicationRequestsComponent;
