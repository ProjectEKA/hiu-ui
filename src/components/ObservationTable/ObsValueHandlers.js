import { getConceptDisplay } from "../common/HealthInfo/FhirResourcesUtils";
/* istanbul ignore file */
const obsProperties = [
  {
    key: 'valueQuantity',
    getValue(o) {
      return `${o.valueQuantity.value.toString()  } ${  o.valueQuantity.unit}`;
    },
  },
  {
    key: 'valueCodeableConcept',
    getValue(o) {
      return getConceptDisplay(o.valueCodeableConcept);
    },
  },
  {
    key: 'valueString',
    getValue(o) {
      return o.valueString;
    },
  },
  {
    key: 'valueBoolean',
    getValue(o) {
      return o.valueBoolean.toString();
    },
  },
  {
    key: 'valueDateTime',
    getValue(o) {
      return o.valueDateTime.toString();
    },
  },
  {
    key: 'valueInteger',
    getValue(o) {
      return o.valueInteger;
    },
  },
  {
    key: 'valuePeriod',
    getValue(o) {
      return (
        `${o.valuePeriod.start.toString()}-${o.valuePeriod.end.toString()}`
      );
    },
  },
  {
    key: 'valueRatio',
    getValue(o) {
      return (
        `${o.valueRatio.numerator.toString()
        }:${
          o.valueRatio.denominator.toString()}`
      );
    },
  },
  {
    key: 'valueRange',
    getValue(o) {
      return `${o.valueRange.low.toString()}-${o.valueRange.high.toString()}`;
    },
  },
  {
    key: 'valueTime',
    getValue(o) {
      return o.valueTime.toString();
    },
  },
  {
    key: 'valueSampleData',
    getValue(o) {
      return o.valueSampleData.data;
    },
  }
];

function valueForObs(obs) {
  for (let i = 0; i < obsProperties.length; i++) {
    const prop = obsProperties[i];
    if (Object.prototype.hasOwnProperty.call(obs, prop.key)) {
      return prop.getValue(obs);
    }
  }
  return undefined;
}

export default valueForObs;
