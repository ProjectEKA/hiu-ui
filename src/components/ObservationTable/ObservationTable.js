import React from 'react';
import * as PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import valueForObs from './ObsValueHandlers';
import TableStyles from '../common/Styles/Table.style';
import { formatDateString, getConceptDisplay } from '../common/HealthInfo/FhirResourcesUtils';
import {getConceptText} from "./ObservationTableHelperFunctions";

export const Components = ({ components }) => (components
  ? components.map((component) => (
    <li style={{ backgroundColor: 'primary' }} key={component.valueString}>
      <span>
        {getConceptText(component.code)}
        {' '}
        :
        {' '}
      </span>
      <span>{valueForObs(component)}</span>
    </li>
  ))
  : null);

const ObservationTable = ({ data }) => {
  function extractInterpretation(entry) {
    return entry.interpretation ? entry.interpretation[0].text : '';
  }

  return data && data.length !== 0 ? (
    <TableStyles>
      <TableContainer className="table-container" component={Paper}>
        <Table className="table" aria-label="simple table">
          <TableHead>
            <TableRow className="table-head">
              <TableCell className="header" colSpan={4}>
                Observation:
              </TableCell>
            </TableRow>
            <TableRow className="table-head">
              <TableCell align="left">Date</TableCell>
              <TableCell align="left">Observation</TableCell>
              <TableCell align="left">Value</TableCell>
              <TableCell align="left">Status and Interpretation</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {data.map((entry) => (
              <TableRow
                key={`${entry.status}-${entry.id}`}
                className={entry.id ? 'children-row' : 'parent-row'}
              > 
                <TableCell className="table-cell">
                  {entry.effectiveDateTime
                    ? formatDateString(entry.effectiveDateTime, true)
                    : ''}
                </TableCell>
                <TableCell className="table-cell">{getConceptDisplay(entry.code)}</TableCell>
                <TableCell className="table-cell">
                  {valueForObs(entry)}
                  <ul>
                    <Components components={entry.component} />
                  </ul>
                </TableCell>
                <TableCell className="table-cell">
                  <div>{entry.status}</div>
                  {extractInterpretation(entry)}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </TableStyles>
  ) : (
    <div />
  );
};

ObservationTable.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({})),
};

ObservationTable.defaultProps = {
  data: [],
};

export default ObservationTable;
