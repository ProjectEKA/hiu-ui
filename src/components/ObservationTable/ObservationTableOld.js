import React, { useEffect, useState } from "react";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import CircularProgress from "@material-ui/core/CircularProgress";
import ObservationTableStyles from "./ObservationTable.style";
import {
  generateObservableEntity,
  generateObservableEntityValue,
  generateObservableEntityStatusInterpretation,
  generateRowsForMembers
} from "./ObservationTableHelperFunctions";
import IconButton from "@material-ui/core/IconButton";
import { ArrowDropDown, ArrowRight } from "@material-ui/icons";

const ObservationMembers = ({ members, close }) => {
  return members
    ? members.map((member, i) => (
        <TableRow
          className={close ? "close" : ""}
          style={{ backgroundColor: "primary" }}
          key={i}
        >
          <TableCell></TableCell>
          <TableCell></TableCell>
          <TableCell>{generateObservableEntity(member)}</TableCell>
          <TableCell>{generateObservableEntityValue(member)}</TableCell>
          <TableCell>
            {generateObservableEntityStatusInterpretation(member)}
          </TableCell>
        </TableRow>
      ))
    : null;
};

const ObservationTable = ({ loadHealthData, healthInfo, consentRequestId }) => {
  const [close, setClose] = useState(0);
  useEffect(() => {
    loadHealthData(consentRequestId);
  }, []);

  function extractObservations(healthInfo) {
    if (healthInfo && healthInfo.entries) {
      const ObservationData = healthInfo.entries[0].data;
      if (ObservationData) {
        const Observations = ObservationData.entry.filter(
          item => item.resource.resourceType === "Observation"
        );
        return Observations;
      }
    }
    return undefined;
  }

  const Observations = extractObservations(healthInfo);
  const Members = Observations && generateRowsForMembers(Observations);

  return Observations ? (
    <ObservationTableStyles>
      <TableContainer component={Paper}>
        <Table className="observation-table" aria-label="simple table">
          <TableHead>
            <TableRow className="table-head">
              <TableCell></TableCell>
              <TableCell align="left">Date</TableCell>
              <TableCell align="left">Observation</TableCell>
              <TableCell align="left">Value</TableCell>
              <TableCell align="left">Status and Interpretation</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow key={Observations.id}>
              <TableCell>
                {Members ? (
                  close ? (
                    <IconButton
                      className="expand-button"
                      size="small"
                      onClick={() => setClose(!close)}
                      color="primary"
                    >
                      <ArrowRight />
                    </IconButton>
                  ) : (
                    <IconButton
                      className="expand-button"
                      size="small"
                      onClick={() => setClose(!close)}
                      color="primary"
                    >
                      <ArrowDropDown />
                    </IconButton>
                  )
                ) : null}
              </TableCell>
              <TableCell align="left">
                {Observations[0].resource.effectiveDateTime}
              </TableCell>
              <TableCell align="left">
                {generateObservableEntity(Observations[0].resource)}
              </TableCell>
              <TableCell align="left">
                {generateObservableEntityValue(Observations[0].resource)}
              </TableCell>
              <TableCell align="left">
                {generateObservableEntityStatusInterpretation(
                  Observations[0].resource
                )}
              </TableCell>
            </TableRow>
            <ObservationMembers members={Members} close={close} />
          </TableBody>
        </Table>
      </TableContainer>
    </ObservationTableStyles>
  ) : (
    <ObservationTableStyles>
      <div className="loader-container">
        <CircularProgress
          id="loader"
          className="loader"
          variant="indeterminate"
          disableShrink
          size={24}
          thickness={4}
        />
      </div>
    </ObservationTableStyles>
  );
};

export default ObservationTable;
