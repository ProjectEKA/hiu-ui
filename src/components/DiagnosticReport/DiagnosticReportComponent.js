import React from 'react';
import * as PropTypes from 'prop-types';
import TableContainer from '@material-ui/core/TableContainer';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import DiagnosticReportComponentStyles from './DiagnosticReportComponent.style';
import getNestedObject from '../../utils/getNestedObject';
import ObservationTable from '../ObservationTable/ObservationTable';
import { formatDateString } from '../common/HealthInfo/FhirResourcesUtils';
import AttachmentLink from '../AttachmentLink';
import Config from "../../Config";

const DiagnosticReportComponent = ({ data, consentReqId }) => {
  const performerArray = [];

  function extractPerformer(entry) {
    if (entry.performer) {
      entry.performer.forEach((performer) => performerArray.push(performer.display));
      return performerArray;
    }
    return undefined;
  }

  const renderPresentedForm = (entry) => (entry.presentedForm ? (
    <div>
      <span>Diagnostic report links and attachments : </span>
      <ul>
        {entry.presentedForm.map((link) => (
          <li key={link.url}>
            <AttachmentLink attachmentPath={link.url} consentReqId={consentReqId} linkTitle={link.title ? link.title : 'Link'} />
          </li>
        ))}
      </ul>
    </div>
  ) : (
    <div />
  ));

  function getResultsList(results, resourceType) {
    const referenceList = [];
    if (results) { 
      results.forEach((result) => {
        if (result.targetResource)  {
          referenceList.push(result.targetResource);
        }
      });
    }
    return referenceList.filter((ref) => (ref.resourceType.toLowerCase() === resourceType.toLowerCase()));
  }

  const renderObservations = (entry) => {
    const ObsList = getResultsList(
      getNestedObject(entry, 'result'),
      'Observation',
    );

    return <ObservationTable data={ObsList} />;
  };

  const getDicomAddress = (entry) => {
    let imageUrl;
    const imagingStudyList = getResultsList(
        getNestedObject(entry, 'imagingStudy'),
        'ImagingStudy',
    );
    if (imagingStudyList) {
      imagingStudyList.forEach((imagingStudyObj) => {
        const endpointList = getResultsList(
            getNestedObject(imagingStudyObj,'endpoint'),
            'Endpoint',
        );
        if (endpointList) {
          endpointList.forEach((endpointObj) => {
            imageUrl = endpointObj.address;
          });
        }
      });
    }
    return imageUrl;
  }

  function getMediaList(results, resourceType) {
    const referenceList = [];
    if (results) {
      results.forEach((result) => {
        const mediaLinkObject = {
          display: result.link.display ? result.link.display : 'Media Link',
          url: result.link.targetResource.content.url,
          targetResource: result.link.targetResource,
        };
        referenceList.push(mediaLinkObject);
      });
    }

    return referenceList.filter((ref) => ref.targetResource.resourceType === resourceType);
  }

  function generateImageUrl(url) {
    const urlArray = url.split('/');
    const StudyInstanceUID = urlArray.slice(-1).pop();
    const dicomCtx = btoa(Config.DICOM_SERVER_PATH);

    const dicomUrl = `${window.location.origin
    }/viewer/${
      StudyInstanceUID
    }?dicomCtx=${
      dicomCtx}`;
    return dicomUrl;
  }

  const renderMedia = (entry) => {
    const MediaList = getMediaList(getNestedObject(entry, 'media'), 'Media');
    return MediaList && MediaList.length !== 0 ? (
      <div>
        <span>Associated media : </span>
        <ul>
          {MediaList.map((link) => (
            <li key={link.url}>
              <a
                href={generateImageUrl(link.url)}
                target="_blank"
                rel="noopener noreferrer"
              >
                {link.display}
              </a>
            </li>
          ))}
        </ul>
      </div>
    ) : (
      <div />
    );
  };

  return data && data.length !== 0 ? (
    data.map((entry) => (
      <DiagnosticReportComponentStyles key={entry.effectiveDateTime}>
        <TableContainer
          className="diagnostic-report-table-container"
          component={Paper}
        >
          <Typography
            className="diagnostic-report-header"
            variant="h6"
            component="h6"
          >
            Diagnostic report :
            {' '}
            {entry.code ? entry.code.text : ''}
          </Typography>
          <div className="diagnostic-report">
            <ul className="report-details-list">
              <li>
                <span>Date: </span>
                {entry.effectiveDateTime
                  ? formatDateString(entry.effectiveDateTime)
                  : '-'}
              </li>
              <li>
                <span>Status: </span>
                {entry.status}
              </li>
              <li>
                <span>Performer: </span>
                {extractPerformer(entry)}
              </li>
              { entry.conclusion && (
              <li>
                <span>
                  Conclusion:
                  {entry.conclusion}
                </span>
              </li>
               )}
              { getDicomAddress(entry) && (
              <li>
                <span> 
                  {' '}
                  <a href={getDicomAddress(entry)}>DICOM image</a>
                  {' '}
                </span>
              </li>
            )}
            </ul>
            {renderObservations(entry)}
            {renderPresentedForm(entry)}
            {renderMedia(entry)}
          </div>
        </TableContainer>
      </DiagnosticReportComponentStyles>
    ))
  ) : (
    <div />
  );
};

DiagnosticReportComponent.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    code: PropTypes.shape({ text: PropTypes.string }),
    status: PropTypes.string,
    effectiveDateTime: PropTypes.string,
    performer: PropTypes.arrayOf(PropTypes.shape({ display: PropTypes.string })),
    presentedForm: PropTypes.arrayOf(PropTypes.shape({
      url: PropTypes.string,
      title: PropTypes.string,
    })),
  })),
  consentReqId: PropTypes.string,
};

DiagnosticReportComponent.defaultProps = {
  data: [],
  consentReqId: null,
};

export default DiagnosticReportComponent;
