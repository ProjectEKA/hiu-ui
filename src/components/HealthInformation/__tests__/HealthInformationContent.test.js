import React from 'react';
import { shallow } from 'enzyme';
import CCRDocument from '../../Composition/CCRDocument';
import ObservationTable from '../../ObservationTable/ObservationTable';
import DiagnosticReportComponent from '../../DiagnosticReport/DiagnosticReportComponent';
import MedicationRequestsComponent from '../../Medication/MedicationRequestsComponent';
import HealthInformationContent from '../HealthInformationContent.view';
import DischargeSummary from "../../DischargeSummary/DischargeSummary.view";

describe('HealthInformationContent', () => {
  const mockData = [
    {
      resourceType: 'MedicationRequest',
      status: 'active',
      intent: 'order',
      medicationReference:
                {
                  reference: 'Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2',
                  display: 'IV antibiotics',
                },
      subject:
                {
                  display: 'Shriya',
                },
      authoredOn: '2020-01-22',
      dosageInstruction: [
        {
          sequence: 1,
          text: '22/01/2020 to 07/02/2020',
          timing: {
            code: {
              text: '15 days',
            },
          },
        },
      ],
    },
  ];
  const wrapper = shallow(
    <HealthInformationContent
      consentReqId="123"
      hipName="Max Health Care"
      data={mockData}
      title="Discharge Summary"
      startDate="2018-04-01T15:30:10+01:00"
      endDate="2018-04-10T15:30:10+01:00"
    />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render CCRDocument with composition data', () => {
    mockData[0].resourceType = 'Composition';
    wrapper.setProps({ data: mockData });
    expect(wrapper.find(CCRDocument).props().compositionData).toEqual(mockData);
    expect(wrapper.find(CCRDocument).props().consentReqId).toEqual('123');
  });

  it('should render ObservationTable with data', () => {
    mockData[0].resourceType = 'Observation';
    wrapper.setProps({ data: mockData });
    expect(wrapper.find(ObservationTable).props().data).toEqual(mockData);
  });

  it('should render DiagnosticReportComponent with data', () => {
    mockData[0].resourceType = 'DiagnosticReport';
    wrapper.setProps({ data: mockData });
    expect(wrapper.find(DiagnosticReportComponent).props().data).toEqual(mockData);
  });

  it('should render with no data', () => {
    wrapper.setProps({ data: undefined });
    expect(wrapper.find(CCRDocument).props().compositionData).toEqual([]);
    expect(wrapper.find(ObservationTable).props().data).toEqual([]);
    expect(wrapper.find(DiagnosticReportComponent).props().data).toEqual([]);
    expect(wrapper.find(MedicationRequestsComponent).props().medicationRequests).toEqual([]);
  });
});
