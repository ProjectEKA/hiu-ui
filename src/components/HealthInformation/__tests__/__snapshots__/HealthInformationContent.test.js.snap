// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthInformationContent should render properly 1`] = `
<styled.div>
  <div
    className="hip-health-info-container"
  >
    <WithStyles(ForwardRef(Typography))
      className="header"
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      Max Health Care
    </WithStyles(ForwardRef(Typography))>
    <CCRDocument
      compositionData={Array []}
      consentReqId="123"
    />
    <ObservationTable
      data={Array []}
    />
    <DiagnosticReportComponent
      consentReqId="123"
      data={Array []}
    />
    <ConditionsComponent
      conditionList={Array []}
    />
    <MedicationRequestsComponent
      medicationRequests={
        Array [
          Object {
            "authoredOn": "2020-01-22",
            "dosageInstruction": Array [
              Object {
                "sequence": 1,
                "text": "22/01/2020 to 07/02/2020",
                "timing": Object {
                  "code": Object {
                    "text": "15 days",
                  },
                },
              },
            ],
            "intent": "order",
            "medicationReference": Object {
              "display": "IV antibiotics",
              "reference": "Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2",
            },
            "resourceType": "MedicationRequest",
            "status": "active",
            "subject": Object {
              "display": "Shriya",
            },
          },
        ]
      }
    />
    <DocumentReferenceComponent
      consentReqId="123"
      data={Array []}
      enclosed={false}
    />
  </div>
</styled.div>
`;
